### PR TITLE
#151980454 implement sorting and filtering

### PR DESCRIPTION
**Pull request by aimeedyki**
#### What does this PR do?

Allows a user to filter and sort product search results by certain Price, date and vendor.

####Description of Task to be completed?

- filter search results by price range
- filter search result by vendor
- sort search result by date and price

#### How should this be manually tested?

- Click on the search icon
- Search for a product
- select a filter or sort criteria

#### Any background context you want to provide?

search results could not be filtered to further refine a search

#### What are the relevant pivotal tracker stories?

[https://www.pivotaltracker.com/story/show/151980454](https://www.pivotaltracker.com/story/show/151980454)

#### Screenshots (if appropriate)
screen shot 2017-11-09 at 5 32 39 pm
screen shot 2017-11-09 at 5 33 32 pm

### CODE REVIEW

**Comment by fidelisojeah** [comment]
- A little overlap here
screen shot 2017-11-10 at 11 36 56 am

- Not sure this sort/filter panel is useful when admin is searching Accounts or Orders
screen shot 2017-11-10 at 11 49 33 am
screen shot 2017-11-10 at 11 44 04 am

File: imports/plugins/included/ui-search/client/templates/searchModal/searchModal.js [hunk 1]
```diff
@@ -27,10 +27,57 @@ Template.searchModal.onCreated(function () {
...
+
+  // filter search by price range
+  const filterByPrice = (products, price1, price2) => {
+    return _.filter(products, (product) => {
+      if (product.price) {
+        const maxPrice = parseFloat(product.price.max);
+        const minPrice = parseFloat(product.price.min);
+        const queryMaxPrice = parseFloat(price2);
+        const queryMinPrice = parseFloat(price1);
+        if (minPrice >= queryMinPrice && maxPrice <= queryMaxPrice) {
```
Comment (femisenjobi) [inline imports/plugins/included/ui-search/client/templates/searchModal/searchModal.js]: Try this instead
```
if ((minPrice >= queryMinPrice && minPrice <= queryMaxPrice) || (maxPrice >= queryMinPrice && maxPrice <= queryMaxPrice)) {
...
}
```

File: imports/plugins/included/ui-search/client/templates/productSearch/productSort.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                  <option value="{{vendor}}">{{vendor}}</option>
+                {{/each}}
+          </select>
+        </div>
+    <div class="col-md-5 slider-fix">
+      <div>
+          <span><b>Price range: </b></span>
+      </div>
+      <div class="row price-fix">
+        <input class="col-md-2" id="price-filter1" type="text" value="{{sliderMin}}"/>
```
Comment (femisenjobi) [inline imports/plugins/included/ui-search/client/templates/productSearch/productSort.html]: Use `type="number"`

File: imports/plugins/included/ui-search/client/templates/productSearch/productSort.html [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        <div class="col-md-7 slider-fix" id="slider"></div>
+        <input class="col-md-2" id="price-filter2" type="text" value="{{sliderMax}}"/>
```
Comment (femisenjobi) [inline imports/plugins/included/ui-search/client/templates/productSearch/productSort.html]: Use `type="number"` here too

**Review by femisenjobi — CHANGES_REQUESTED** [review]
Thanks for making the changes, I have 2 comments
- the conditional that returns the products seem restricting. Once my minimum filter is above the minimum price, the product is not included. If that happens and the max price is still below my max query, the product should still be suggested. I suggested a better conditional that solves that edge case
- Secondly, it would be great to use `type="number"` for the inputs. At the moment, it takes text and does not respond to arrow up/down buttons. Using `type="number"` gives us that feature and validation

File: imports/plugins/included/ui-search/client/templates/productSearch/productSort.html (hunk 2 above)
Comment (femisenjobi) [inline imports/plugins/included/ui-search/client/templates/productSearch/productSort.html]: also use `class="col-xs-2"` instead

File: imports/plugins/included/ui-search/client/templates/productSearch/productSort.html [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                {{/each}}
+          </select>
+        </div>
+    <div class="col-md-5 slider-fix">
+      <div>
+          <span><b>Price range: </b></span>
+      </div>
+      <div class="row price-fix">
+        <input class="col-md-2" id="price-filter1" type="text" value="{{sliderMin}}"/>
+        <div class="col-md-7 slider-fix" id="slider"></div>
```
Comment (femisenjobi) [inline imports/plugins/included/ui-search/client/templates/productSearch/productSort.html]: Also, use class="col-xs-7" instead

File: imports/plugins/included/ui-search/client/templates/productSearch/productSort.html (hunk 3 above)
Comment (femisenjobi) [inline imports/plugins/included/ui-search/client/templates/productSearch/productSort.html]: Also, use class="col-xs-2" instead

**Review by femisenjobi — CHANGES_REQUESTED** [review]
Also, use
- col-xs-2 instead of col-md-2 (for the input fields)
- col-xs-7 instead of col-md-7 (for the slider)
This makes the input and slider appear on the same row even in mobile devices (md for laptops; sm for tabs; xs for phones)

**Comment by aimeedyki** [comment]
Thanks, changes have been implemented.

**Review by femisenjobi — APPROVED** [review]

**Pull request merged**